src/frontend/components/BasicNoteCard/BasicNoteCard.js

```diff
@@ -42,36 +74,60 @@ const BasicNoteCard = ({ note, setNotes }) => {
+                            onBlur={() => { dispatchBasicNote({ type: "TITLETOGGLE", payload: true }); titleEdit(newTitle) }}
+                            onChange={(e) => dispatchBasicNote({ type: "NEWTITLE", payload: e.target.value })}
+                        />}
                     <div onClick={() => pinToggle()}>
                         {pinned ? <span className="material-icons pinned" title="Unpin">push_pin</span> :
                             <span className="material-icons unpinned" title="Pin">push_pin</span>}
                     </div>
                 </div>
-                <p className="note__copy secondary__font" dangerouslySetInnerHTML={{ __html: content }}></p>
+                <div className="flex--column note__copyContainer" onBlur={(e) => { handleContentEditBlur(e); contentEdit(newContent) }}>
+                    {contentToggle ? <p className="note__copy secondary__font" dangerouslySetInnerHTML={{ __html: newContent }} onClick={() => dispatchBasicNote({ type: "CONTENTTOGGLE", payload: false })}></p> :
+                        <BasicNoteEditor content={newContent} setValue={dispatchBasicNote} />}</div>
             </div>
             {label && <div className="note__label font__secondary" onClick={() => labelPicker("")}>
                 {label}
             </div>}
             <div className="flex--row note__optionpicker">
-                <span className="material-icons colorpicker__btn" onClick={() => { setLabellist(false); setColorlist(!colorlist) }} title="Change Color">palette</span>
+                <span className="material-icons colorpicker__btn" onClick={() => { dispatchBasicNote({ type: "LABELLIST", payload: false }); dispatchBasicNote({ type: "COLORLIST", payload: !colorlist }) }} title="Change Color">palette</span>
                 {colorlist && <div className="color__list flex--row">
-                    <div className="color1" onClick={() => { setColorlist(false); colorPicker("note--color1") }}></div>
-                    <div className="color2" onClick={() => { setColorlist(false); colorPicker("note--color2") }}></div>
-                    <div className="color3" onClick={() => { setColorlist(false); colorPicker("note--color3") }}></div>
-                    <div className="color4" onClick={() => { setColorlist(false); colorPicker("note--color4") }}></div>
+                    <div className="color1" onClick={() => { dispatchBasicNote({ type: "COLORLIST", payload: false }); colorPicker("note--color1") }}></div>
+                    <div className="color2" onClick={() => { dispatchBasicNote({ type: "COLORLIST", payload: false }); colorPicker("note--color2") }}></div>
+                    <div className="color3" onClick={() => { dispatchBasicNote({ type: "COLORLIST", payload: false }); colorPicker("note--color3") }}></div>
+                    <div className="color4" onClick={() => { dispatchBasicNote({ type: "COLORLIST", payload: false }); colorPicker("note--color4") }}></div>
```

Comment:
This is repetitive code. you can make an array of colors and then use map

Reply:
yup did that rn in the next commit